lib/internal/blob.js

```diff
@@ -56,17 +61,35 @@ const {
 const kHandle = Symbol('kHandle');
 const kType = Symbol('kType');
 const kLength = Symbol('kLength');
+const kArrayBufferPromise = Symbol('kArrayBufferPromise');
 
 const disallowedTypeCharacters = /[^\u{0020}-\u{007E}]/u;
 
 let Buffer;
+let ReadableStream;
+let URL;
+
+
+// Yes, lazy loading is annoying but because of circular
+// references between the url, internal/blob, and buffer
+// modules, lazy loading here makes sure that things work.
+
+function lazyURL(id) {
+  URL ??= require('url').URL;
+  return new URL(id);
+}
 
 function lazyBuffer() {
-  if (Buffer === undefined)
-    Buffer = require('buffer').Buffer;
+  Buffer ??= require('buffer').Buffer;
```

Comment:
Why don't we just do:
```diff
-    source = lazyBuffer().from(`${source}`, encoding);
-    source = new TextEncoder().encode(source);
```
...and get rid of Buffer?

Reply:
That's done in a separate PR